packages/cli/src/credentials/credentials.service.ee.ts

```diff
@@ -0,0 +1,42 @@
+/* eslint-disable import/no-cycle */
+import { Db } from '..';
+import { CredentialsService } from './credentials.service';
+import { CredentialsEntity } from '../databases/entities/CredentialsEntity';
+import { SharedCredentials } from '../databases/entities/SharedCredentials';
+import { User } from '../databases/entities/User';
+import { RoleService } from '../role/role.service';
+
+export class EECreditentialsService extends CredentialsService {
+	static async isOwned(
+		user: User,
+		credentialId: string,
+	): Promise<{ ownsCredential: boolean; credential?: CredentialsEntity }> {
+		const sharedCredentials = await this.getSharedCredentials(user.id, credentialId, [
+			'credentials',
+		]);
+
+		return sharedCredentials
+			? { ownsCredential: true, credential: sharedCredentials.credentials }
```

Comment:
yes I also struggled a bit with it, but then decided how it is often done in elixir. You receive a tuple (or boolean) as  a response that you can check against and if it is true you know you have the data and can continue using it without another trip to the DB. Since the function's signature is showing and describing the return value I thought it is not an issue but rather adds to simplicity, wdyt?